torch/_dynamo/side_effects.py

```diff
@@ -75,6 +75,26 @@ def __init__(self, id_to_variable=None, store_attr_mutations=None, keepalive=Non
         self.store_attr_mutations = store_attr_mutations or collections.OrderedDict()
         self.keepalive = keepalive or []
 
+    def __eq__(self, other: object) -> bool:
+        assert isinstance(other, SideEffects)
+        # NB: do NOT test keepalive
+        return (
+            self.id_to_variable == other.id_to_variable
+            and self.store_attr_mutations == other.store_attr_mutations
+        )
+
+    def diff(self, other: "SideEffects") -> Optional[str]:
+        if self.id_to_variable != other.id_to_variable:
+            sk_itv = self.id_to_variable.keys()
+            ok_itv = other.id_to_variable.keys()
```

Comment:
this diff *does* care about keys, but the one above does not. We should be consistent if we are going to make a `diff` a pattern a la `eq`

Reply:
this is because i was lazy. The current code is sound, but it won't give you a nice error message saying that the keys differ in the store_attr_mutations case. I can make it symmetric but when I was debugging bugs I only needed the `id_to_variable` case.